test/jdk/java/awt/Dialog/DialogLocationTest.java

```diff
@@ -0,0 +1,182 @@
+        int height, width, insetX, insetY;
+        Point curLoc;
+        int i;
+
+        Rectangle screen = GraphicsEnvironment
+                .getLocalGraphicsEnvironment()
+                .getMaximumWindowBounds();
+        height = screen.height;
+        width = screen.width;
+        insetX = screen.x;
+        insetY = screen.y;
+
+        String message = "Failed on iteration %d expect:[%d,%d] "
+                + "reported:[%d,%d] diff:[%d,%d]";
+
+        for (i = 0; i < 100; i++) {
+            newX = random.nextInt(insetX, width - 300);
+            newY = random.nextInt(insetY, height - 400);
+
+            if (newX == 0 && newY == 0) {
+                i--;
+                continue;
+            }
+
+            waitingForEvent = true;
+
+            EventQueue.invokeLater(() -> {
+                dialog.setLocation(newX, newY);
+            });
+
+            while (waitingForEvent) {
+                Thread.yield();
+            }
+
+            curLoc = dialog.getLocation();
+            if (curLoc.x != newX || curLoc.y != newY) {
+                count++;
+                System.out.println(message.formatted(i, newX, newY,
+                        curLoc.x, curLoc.y, curLoc.x - newX, curLoc.y - newY));
+                System.out.flush();
+            }
+        }
+
+        if (count > 0) {
+            throw new RuntimeException("Dialog Location was set incorrectly");
+        }
+    }
+
+    public class CL extends ComponentAdapter {
+        int lastX, lastY;
+        String message = "Failed in componentMoved() expect:[%d,%d]"
+                + " reported: [%d,%d] diff [%d,%d]";
+
+        public void componentMoved(ComponentEvent e) {
+            if (e.getComponent() == my_dialog) {
+                Point eventLoc = e.getComponent().getLocation();
+                if (lastX != eventLoc.x || lastY != eventLoc.y) {
+                    lastX = eventLoc.x;
+                    lastY = eventLoc.y;
+                    if (newX != 0 && newY != 0 && (eventLoc.x != newX || eventLoc.y != newY)) {
+                        count++;
+                        System.out.println(message.formatted(newX, newY,
+                                eventLoc.x, eventLoc.y,
+                                eventLoc.x - newX, eventLoc.y - newY));
+                        System.out.flush();
+                    }
+                    waitingForEvent = false;
+                }
+            }
+        }
+    }
+
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException {
+        Frame frame = new Frame("DialogLocationTest");
+        try {
+            DialogLocationTest test = new DialogLocationTest();
+            EventQueue.invokeAndWait(() -> {
+                frame.add(test);
+                test.init();
+                frame.setVisible(true);
+            });
+            test.start();
+        } finally {
+            EventQueue.invokeLater(() -> {
```

Comment:
invokeLater may not be executed before the test stops by the jtreg, EDT could be killed at that point. It is better to use invokeAndWait or somehow make sure it will ends.

Reply:
> invokeLater may not be executed before the test stops by the jtreg, EDT could be killed at that point. It is better to use invokeAndWait or somehow make sure it will ends.

Totally agree.